### PR TITLE
[Bug](serde) fix serialize column to jsonb when meet boolean and decimal_v3

### PR DESCRIPTION
**Pull request by eldenmoon**
…mal_v3

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.73 seconds
 stream load tsv:          428 seconds loaded 74807831229 Bytes, about 166 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230424120636_clickbench_pr_134066.html

**Review by yiguolei — APPROVED** [review]
LGTM

File: be/src/vec/data_types/serde/data_type_number_serde.h [hunk 1]
```diff
@@ -210,7 +210,7 @@ void DataTypeNumberSerDe<T>::write_one_cell_to_jsonb(const IColumn& column,
                                                      int row_num) const {
     result.writeKey(col_id);
     StringRef data_ref = column.get_data_at(row_num);
-    if constexpr (std::is_same_v<T, Int8>) {
+    if constexpr (std::is_same_v<T, Int8> || std::is_same_v<T, UInt8>) {
         int8_t val = *reinterpret_cast<const int8_t*>(data_ref.data);
```
Comment (yiguolei) [inline be/src/vec/data_types/serde/data_type_number_serde.h:214]: if the value is uint8, but cast it to int8_t at line 214, maybe something lost?

**Review by yiguolei — CHANGES_REQUESTED** [review]
LGTM is mistake

File: be/src/vec/data_types/serde/data_type_number_serde.h (hunk 1 above)
Comment (eldenmoon) [reply be/src/vec/data_types/serde/data_type_number_serde.h:214]: doris has no uint8 type, uint8 is for boolean, so lost is not possible here

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**